### PR TITLE
Bumping neurodamus to 2.5.3

### PR DESCRIPTION
**Pull request by ferdonline**
Neurodamus 2.5.3 provides a critical fix for spike delivery from replay and other circuits.

Bonus:
 - Clean older unimportant versions
 - Add optional dependencies, so environment is prepared for any workload

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: Is this now used with py-neurodamus?

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-neurodamus/package.py:32]: Same here? python interface now used directly?

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: typically we keep older versions but wouldn't accumulate 30 different versions that no one going to install in practice.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: Seems reasonable to keep the latest 1.x release and purge the rest. IIRC, at some point we should be able to use git references with Spack. Then having a few earlier versions for dependencies makes sense, and no need to keep 10s of others.

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: Used only for local to global positions translation (Baudouin work), and therefore is not even a mandatory dependency in `setup.py`

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:32]: In some use cases, like ngv or nodesets, yes, we started using libsonata directly. But again, it's an optional dependency, i.e. neurodamus runs fine without it, until the point it absolutely requires (late import)

Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:32]: We are registering dependency here so that the main deployed neurodamus works for all workloads.

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: So you agree with the current set? I still kept like 10 versions which have a chance of being built, I think all others we can forget (and if ever needed we can checkout spack in another revision)

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:31]: Yes. Thinning out a little at this point is good.

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (pramodk) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:32]: yeah, its fine. I worry about about the dependencies when people build stuff locally and then all world starts building :)

**Review by pramodk — APPROVED** [review]
LGTM - once Armando's test passes then we can merge it

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (alexsavulescu) [inline var/spack/repos/builtin/packages/py-neurodamus/package.py:35]: Let's also put this in for 2.5.3 (needed for morphio_wrapper)

**Comment by ferdonline** [comment]
@pramodk It seems to fix https://bbpteam.epfl.ch/project/issues/browse/CNEUR-372 (see Armando comment)
@alexsavulescu Even though we agreed on declaring morphio as dependency, I now think it's better to have it for next release. It makes things match better, since 2.5.3 doesn't require it yet. Like that we can also merge this right away.

**Comment by pramodk** [comment]
LGTM for merge!

File: var/spack/repos/builtin/packages/py-neurodamus/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-neurodamus/package.py:32]: True it could be an optional dependency also here 🤔 but then in our deployment script we would have to enable the full-dependency variant

**Pull request merged**
